hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/JDBCExecutor.java

```diff
@@ -129,10 +129,11 @@ public Map<String, String> getTableSchema(String tableName) {
     ResultSet result = null;
     try {
       DatabaseMetaData databaseMetaData = connection.getMetaData();
-      result = databaseMetaData.getColumns(null, databaseName, tableName, null);
+      String catalog = connection.getCatalog();
+      result = databaseMetaData.getColumns(catalog, databaseName, tableName, "%");
```

Comment:
Not very familiar with JDBC, can you elaborate the effect of changes for line 133?

Reply:
@danny0405 Thank you very much for your help in reviewing the code!

For this part of JDBC, we use HiveJDBC. Before I changed it, I referred to some codes of Hive.


HiveDatabaseMetaData#getColumns
```
  public class HiveDatabaseMetaData implements DatabaseMetaData {
  public ResultSet getColumns(String catalog, String schemaPattern,
      String tableNamePattern, String columnNamePattern) throws SQLException {}
  .....
  }
```

The call stack of the code is as follows:
```
Hive 
  \-- CLIService#getColumns
      \-- HiveSessionImpl#getColumns
           \-- GetColumnsOperation#runInternal
```

By reading GetColumnsOperation#runInternal, we can find that `catalogName` has no obvious effect, so we can set it to null,  But Hive's Beeline code directly uses `HiveConnetion's getCatalog`,  So I also follow Hive's usage.

If columnNamePattern is null, it means to get all fields, but this is not conducive to reading, % means no filtering, matching all columns, which is easier to understand.